lib/routes/close_channels/coop_closed_channels.dart

```diff
@@ -0,0 +1,241 @@
+                          style: theme.validatorStyle,
+                        )
+                      : const SizedBox(),
+                ],
+              ),
+            )),
+      ),
+      bottomNavigationBar: Padding(
+        padding: const EdgeInsets.only(bottom: 0.0),
+        child: SingleButtonBottomBar(
+          stickToBottom: true,
+          text: texts.close_channels_action_submit,
+          onPressed: (_addressValidated == null) ? null : () => onSubmit(),
+        ),
+      ),
+    );
+  }
+
+  Future _scanBarcode() async {
+    _log.info("Scanning barcode");
+    final texts = context.texts();
+    FocusScope.of(context).requestFocus(FocusNode());
+    await Navigator.pushNamed<String>(context, "/qr_scan").then((barcode) {
+      if (barcode.isEmpty) {
+        _log.info("Received an empty barcode");
+        showFlushbar(
+          context,
+          message: texts.close_channels_qr_code_not_detected,
+        );
+        return;
+      } else {
+        _log.info("Barcode: $barcode");
+      }
+
+      BTCAddressInfo btcInvoice = parseBTCAddress(barcode);
+      setState(() {
+        _addressController.text = btcInvoice.address;
+        _scannerErrorMessage = "";
+      });
+    }, onError: (error) {
+      _log.warning("Scanning barcode error", error);
+      setState(() {
+        _scannerErrorMessage = error.toString();
+      });
+    });
+  }
+
+  Future<bool> _asyncValidate() {
+    _log.info("Validating address: ${_addressController.text}");
+    return _breezLib.validateAddress(_addressController.text).then((data) {
+      _log.info("Address validated: $data");
+      _addressValidated = data;
+      return _formKey.currentState.validate();
+    }).catchError((err) {
+      _log.warning("Validating address error", err);
+      _addressValidated = null;
+      return _formKey.currentState.validate();
+    });
+  }
+
+  void onSubmit() {
+    _asyncValidate().then((validated) async {
+      if (validated) {
+        final texts = context.texts();
+        final themeData = Theme.of(context);
+
+        _formKey.currentState.save();
+        FocusScope.of(context).requestFocus(FocusNode());
+        promptAreYouSure(
+          context,
+          texts.close_channels_title,
+          RichText(
+            text: TextSpan(
+              style: themeData.dialogTheme.contentTextStyle,
+              text: texts.close_channels_confirmation_message,
+              children: [
+                TextSpan(
+                  text: _addressController.text,
+                  style: themeData.dialogTheme.contentTextStyle.copyWith(
+                    fontWeight: FontWeight.bold,
+                  ),
+                ),
+                TextSpan(
+                  text: "?",
+                  style: themeData.dialogTheme.contentTextStyle,
+                ),
+              ],
+            ),
+          ),
+        ).then((ok) async {
+          if (ok) {
+            await closeChannels();
```

Comment:
It looks like we are ignoring the result here. It seems that the API returns information on each channel that was closed (or failed).
I am only worried if for some reason some of the channels weren't able to be closed so perhaps we should reflect that in the UI.
I suggest to involve @kingonly regarding the user experience here.